src/scifem/eval.py

```diff
@@ -0,0 +1,74 @@
+from mpi4py import MPI
+import dolfinx
+import numpy as np
+import numpy.typing as npt
+
+
+def evaluate_function(
+    u: dolfinx.fem.Function, points: npt.ArrayLike, broadcast=True
+) -> npt.NDArray[np.float64]:
+    """Evaluate a function at a set of points.
+
+    Args:
+        u: The function to evaluate.
+        points: The points to evaluate the function at.
+        broadcast: If True, the values will be broadcasted to all processes.
+
+    Returns:
+        The values of the function evaluated at the points.
```

Comment:
If the function is discontinuous, one might get different values from different processes. We should add a note regarding this, and that broadcast takes the largest of those values.